Libraries/Database/SessionExtensions.cs

```diff
@@ -40,18 +40,53 @@ public static class SessionExtensions
                 parameters,
                 out var query,
                 out var messages);
-
-            if (!parserResult)
-            {
-                throw new RawQueryException(
-                    $"Error during parsing the query \"{sql}\" with the following messages: {Environment.NewLine}{string.Join(Environment.NewLine, messages)}",
-                    messages);
-            }
+            ThrowIfParseFailed(parserResult, sql, messages);
 
             return queryExecutor == null
                 ? await transaction.Connection.QueryAsync<TRow>(query, transaction: transaction)
                 : await queryExecutor((query, transaction));
         }
+
+        /// <summary>
+        /// Executes a raw SQL string command in a database-agnostic way by running it through Orchard's <see
+        /// cref="SqlParser"/>.
+        /// </summary>
+        /// <param name="sql">
+        /// The raw SQL string. Doesn't need to use table prefixes or care about SQL dialects.
+        /// </param>
+        /// <param name="parameters">Input parameters passed to the query.</param>
+        /// <param name="transaction">If not <see langword="null"/> it must be an open DB transaction.</param>
+        /// <returns>The number of rows affected.</returns>
+        public static async Task<int> RawExecuteAsync(

```

Comment:
Inkább?
```suggestion
        public static async Task<int> RawExecuteScalarAsync(

```

Reply:
Nope. Ez nem egy int típusú query result, hanem "The number of rows affected."

Reply:
Ja igen. Akkor viszont annak kellene a névnek tükröznie.

Reply:
De tükrözi mert "Raw" + DbConnection metódusnév, pont mint a RawQueryAsync-nél.

Reply:
Legyen Raw[ExecuteNonQuery](https://docs.microsoft.com/en-us/dotnet/api/system.data.sqlclient.sqlcommand.executenonquery?view=dotnet-plat-ext-5.0)?

Reply:
Aha!